src/scalar/scacore_mod.F90

```diff
@@ -1,79 +1,245 @@
 MODULE scacore_mod
     USE core_mod
-    USE ib_mod, ONLY: ftoc, parent, ib
+    USE ib_mod, ONLY: ftoc, parent, ib, gc_t
+    USE flow_mod, ONLY: has_flow
 
     IMPLICIT NONE(type, external)
     PRIVATE
 
+    TYPE :: scalar_bc_t
```

Comment:
Yea... It's there in the code you found.

Personally I would have opted to swap the values of the ityp, because then the default BC become (0, 0.0), i.e. a zero flux condition, and I think just zeros are more logical than a (1, 0.0)... But I decided to keep it this way because you and/or Johannes decided that 0 was a "value" and 1 was a "flux" in the grids.h5 external boundary conditions specification... So it's the same meaning as in the grids.h5 also for the geometries.